pkg/fortify/fpr_to_sarif.go

```diff
@@ -0,0 +1,767 @@
+	RuleID    string  `json:"ruleId"`
+	RuleIndex int     `json:"ruleIndex"`
+	Level     string  `json:"level,omitempty"`
+	Message   Message `json:"message"`
+	/*Locations        []Location        `json:"locations"`
+	CodeFlows        []CodeFlow        `json:"codeFlows"`
+	RelatedLocations []RelatedLocation `json:"relatedLocations"`*/
+	Properties SarifProperties `json:"properties"`
+}
+
+type Message struct {
+	Text string `json:"text,omitempty"`
+}
+
+type SarifProperties struct {
+	InstanceID        string `json:"InstanceID"`
+	InstanceSeverity  string `json:"InstanceSeverity"`
+	Confidence        string `json:"Confidence"`
+	Audited           bool   `json:"Audited"`
+	ToolAuditState    string `json:"ToolAuditState"`
+	ToolAuditMessage  string `json:"ToolAuditMessage"`
+	UnifiedAuditState string `json:"UnifiedAuditState"`
+}
+
+// These structs are relevant to the Tool object
+
+type Tool struct {
+	Driver Driver `json:"driver"`
+}
+
+type Driver struct {
+	Name           string      `json:"name"`
+	Version        string      `json:"version"`
+	InformationUri string      `json:"informationUri,omitempty"`
+	Rules          []SarifRule `json:"rules"`
+	//SupportedTaxonomies []SupportedTaxonomies `json:"supportedTaxonomies"`
+}
+
+type SarifRule struct {
+	Id                   string               `json:"id"`
+	Guid                 string               `json:"guid"`
+	Name                 string               `json:"name,omitempty"`
+	ShortDescription     Message              `json:"shortDescription"`
+	FullDescription      Message              `json:"fullDescription"`
+	DefaultConfiguration DefaultConfiguration `json:"defaultConfiguration"`
+	Relationships        []Relationships      `json:"relationships,omitempty"`
+	Properties           *SarifRuleProperties `json:"properties,omitempty"`
+}
+
+type SupportedTaxonomies struct {
+	Name  string `json:"name"`
+	Index int    `json:"index"`
+	Guid  string `json:"guid"`
+}
+
+type DefaultConfiguration struct {
+	Properties DefaultProperties `json:"properties"`
+	Level      string            `json:"level,omitempty"` //This exists in the template, but not sure how it is populated. TODO.
+}
+
+type DefaultProperties struct {
+	DefaultSeverity string `json:"DefaultSeverity"`
+}
+
+type Relationships struct {
+	Target Target   `json:"target"`
+	Kinds  []string `json:"kinds"`
+}
+
+type Target struct {
+	Id            string        `json:"id"`
+	ToolComponent ToolComponent `json:"toolComponent"`
+}
+
+type ToolComponent struct {
+	Name string `json:"name"`
+	Guid string `json:"guid"`
+}
+
+type SarifRuleProperties struct {
+	Accuracy    string `json:"Accuracy,omitempty"`
+	Impact      string `json:"Impact,omitempty"`
+	Probability string `json:"Probability,omitempty"`
+}
+
+func ConvertFprToSarif(sys System, project *models.Project, projectVersion *models.ProjectVersion, resultFilePath string) error {
+	log.Entry().Debug("Extracting FPR.")
+	_, err := FileUtils.Unzip(resultFilePath, "result/")
```

Comment:
Delete this directory once the parse is complete. Try to unzip it in a tmp directory. It is safe.